core/model/src/test/java/org/eclipse/rdf4j/model/DynamicModelConcurrentModificationAndUpgradeTest.java

```diff
@@ -75,20 +91,23 @@ public boolean contains(Object o) {
 							@Override
 							public boolean hasNext() {
 								try {
-									countDownLatch.await();
+									countDownLatch1.countDown();
+									countDownLatch2.await();
```

Comment:
This latching here should help reduce the chance of this one thread getting executed first.